tests/test_resource_upload.py

```diff
@@ -0,0 +1,165 @@
+    assert os.path.exists(practice_filesystem[f1])
+    assert (copy_resources_s3.get_mime_type(practice_filesystem[f1]) ==
+            "application/json")
+
+
+def test_upload_resources(practice_filesystem, mocker):
+    s3_dir = 'resources'
+    bucket_name = 'test-bucket'
+    sha1_map = copy_resources_s3.new_resource_hashes(
+               practice_filesystem[test_dir])
+    hash_keys = list(sha1_map)
+    s3_client = boto3.client('s3')
+    stubber = botocore.stub.Stubber(s3_client)
+
+    full_key1 = s3_dir + "/" + hash_keys[0]
+    full_key2 = s3_dir + "/" + hash_keys[1]
+    full_key3 = s3_dir + "/" + hash_keys[2]
+
+    stubber.add_client_error('head_object',
+                             service_error_meta={'Code': '404'},
+                             expected_params={
+                                 'Bucket': bucket_name,
+                                 'Key': full_key1
+                              }
+                             )
+    stubber.add_response('put_object', {},
+                         expected_params={
+                            'Body': botocore.stub.ANY,
+                            'Bucket': bucket_name,
+                            'Key': full_key1,
+                            'ContentType': 'application/json'
+                          }
+                         )
+    stubber.add_client_error('head_object',
+                             service_error_meta={'Code': '404'},
+                             expected_params={
+                                'Bucket': bucket_name,
+                                'Key': full_key2
+                                }
+                             )
+    stubber.add_response('put_object', {},
+                         expected_params={
+                            'Body': botocore.stub.ANY,
+                            'Bucket': bucket_name,
+                            'Key': full_key2,
+                            'ContentType': 'application/json'
+                            }
+                         )
+    stubber.add_client_error('head_object',
+                             service_error_meta={'Code': '404'},
+                             expected_params={
+                                'Bucket': bucket_name,
+                                'Key': full_key3
+                                }
+                             )
+    stubber.add_response('put_object', {},
+                         expected_params={
+                            'Body': botocore.stub.ANY,
+                            'Bucket': bucket_name,
+                            'Key': full_key3,
+                            'ContentType': 'application/json'
+                            }
+                         )
+    stubber.activate()
+    mocker.patch("boto3.client", lambda service: s3_client)
+
+    resource_dir = practice_filesystem[test_dir]
+    metadata_dir = practice_filesystem[metadata_file]
+    copy_resources_s3.upload_resources(resource_dir=resource_dir,
```

Comment:
Just to improve code coverage, you can invoke this via `main()` directly if you want with something like:

```python
mocker.patch(
    "sys.argv",
    ["", resource_dir, metadata_dir, bucket, s3_dir]
)
copy_resources_s3.main()

```